core/src/main/scala/kafka/log/LogManager.scala

```diff
@@ -362,7 +401,7 @@ class LogManager(val logDirs: Array[File],
         val dataDir = nextLogDir()
         val dir = new File(dataDir, topicPartition.topic + "-" + topicPartition.partition)
         dir.mkdirs()
-        val log = new Log(dir, config, recoveryPoint = 0L, scheduler, time)
+        val log = new Log(dir, config, logStartOffset = 0L, recoveryPoint = 0L, scheduler, time)
```

Comment:
As of now, the base offset of the first segment of a compacted topic can be always 0. Thus it seems OK to have its logStartOffset as 0. We can consider logStartOffset as the lower bound of the offset of the first message in the partition, but not necessarily the strict lower bound. I couldn't find any use-case that would be broken due to this definition.